### PR TITLE
Update the default Gunicorn API server workers count to one

### PR DESCRIPTION
**Pull request by ssheng**
## Description

Update the default API GUnicorn server workers count to one. Originally, the workers count was automatically determined by the number of available CPU cores.

## Motivation and Context

The automatic determination was not ideal because users were often unaware the large number of workers and large number of workers could result in additional memory overhead from models. Users will still have the ability to set BentoML to automatically determine the number of workers by explicit configuration.

## How Has This Been Tested?

./dev/format.sh
./dev/lint.sh
./ci/unit_test.sh

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [x] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/yatai/deployment/sagemaker/serve [hunk 1]
```diff
@@ -66,4 +66,16 @@ def _serve():
...
+    if "BENTOML_GUNICORN_TIMEOUT" in os.environ:
+        config.override(["api_server", "port"], int(os.environ.get("BENTOML_GUNICORN_TIMEOUT")))
+    if "BENTOML_GUNICORN_NUM_OF_WORKERS" in os.environ:
+        config.override(["api_server", "workers"], int(os.environ.get("BENTOML_GUNICORN_NUM_OF_WORKERS")))
+    container.config.from_dict(config.as_dict())
+
+    from bentoml import service, yatai
+
+    container.wire(packages=[service, yatai])
+
```
Comment (ssheng) [inline bentoml/yatai/deployment/sagemaker/serve:80]: What's the best way to test this? @parano @yubozhao @bojiang

Comment (yubozhao) [reply bentoml/yatai/deployment/sagemaker/serve:80]: We don't have a good way right now.  Deployment tests (e2e_tests) are running offline right now.

We need to refactor/improve the sagemaker deployment with better gunicorn supports anyway. I am fine that we continue to do the basic testings on the e2e test

**Review by yubozhao — APPROVED** [review]

File: bentoml/yatai/deployment/sagemaker/serve [hunk 2]
```diff
@@ -66,4 +66,14 @@ def _serve():
 
 
 if __name__ == '__main__':
+    container = BentoMLContainer()
+    config = BentoMLConfiguration()
+    if "BENTOML_GUNICORN_TIMEOUT" in os.environ:
+        config.override(["api_server", "port"], int(os.environ.get("BENTOML_GUNICORN_TIMEOUT")))
```
Comment (parano) [inline bentoml/yatai/deployment/sagemaker/serve:72]: is this a typo? `s/port/timeout`

File: bentoml/yatai/deployment/sagemaker/serve [hunk 3]
```diff
@@ -32,7 +28,11 @@ def sigterm_handler(nginx_pid, gunicorn_pid):
     sys.exit(0)
 
 
-def _serve():
+@inject
+def _serve(
+    bento_server_timeout: int = Provide[BentoMLContainer.config.api_server.timeout],
+    bento_server_workers: int = Provide[BentoMLContainer.api_server_workers],
```
Comment (parano) [inline bentoml/yatai/deployment/sagemaker/serve:34]: We can also remove the CPU core related calculation in `BentoMLContainer.api_server_workers` right?

File: bentoml/yatai/deployment/sagemaker/serve (hunk 2 above)
Comment (ssheng) [reply bentoml/yatai/deployment/sagemaker/serve:72]: Good catch. This was a typo. We should think about adding unit test for this.

File: bentoml/yatai/deployment/sagemaker/serve (hunk 3 above)
Comment (ssheng) [reply bentoml/yatai/deployment/sagemaker/serve:34]: We we still want to maintain the behavior to automatically determine works if workers is set to `None`?

File: bentoml/yatai/deployment/sagemaker/serve (hunk 2 above)
Comment (ssheng) [reply bentoml/yatai/deployment/sagemaker/serve]: Fixed.

**Comment by parano** [comment]
Thanks, @ssheng, I'm merging this PR now, feel free to address the issue we just discussed in a follow-up PR.

**Pull request merged**